src/pages/PostList.jsx

```diff
@@ -16,13 +20,49 @@ const PostList = () => {
     setSearchQuery(inputValue.trim());
   };
 
+  useEffect(() => {
+    const enrichPosts = async () => {
+      const zipcodes = posts.map(post => post.zip).join(",");
+
+      if (!zipcodes) {
+        setPostsWithCoords(posts);
+        return;
+      }
+
+      try {
+        const locations = await getCoordinatesByZipCodes(zipcodes);
+
+        const updatedPosts = posts.map(post => {
+          const location = locations[post.zip];
+          if (location) {
+            return { ...post, latitude: location.lat, longitude: location.lng };
```

Comment:
You could add the rest of the information in the post object. The `PostCard` component displays the other address information and would benefit from that.

```suggestion
            return { ...post, ...location, latitude: location.lat, longitude: location.lng };
```

If you rename the `lat` and `lng` properties at `getCoordinatesByZipCodes` to `latitude` and `longitude`, you could even do this instead:

```suggestion
            return { ...post, ...location };
```

---

Your branch doesn't contain the latest changes, so I suggest merging `main` into your branch. I did it locally and there wasn't conflicts.

You will see that the `PostCard` component is making a request to zipcodebase. That isn't needed anymore since you are already fetching all the information here 🥇 

If you prefer to do it in another PR, go ahead.